yt/yt/server/queue_agent/queue_static_table_exporter.cpp

```diff
@@ -838,24 +1120,88 @@ void TQueueExporter::Export()
     AlertCollector_->PublishAlerts();
 }
 
-void TQueueExporter::GuardedExport()
+void TQueueStaticTableExporter::GuardedExport(ui64 exportUnixTs)
 {
-    auto config = GetConfig();
+    // XXX(apachee): Throttle inside export task?
+    auto throttler = QueueStaticTableExportManager_->GetExportThrottler();
+    WaitFor(throttler->Throttle(1))
+        .ThrowOnError();
+
+    TQueueExportTaskPtr exportTask;
+    TQueueStaticExportConfigPtr exportConfig;
+    {
+        auto guard = Guard(Lock_);
+
+        exportConfig = ExportConfig_;
+
+        exportTask = New<TQueueExportTask>(
+            ClientDirectory_->GetClientOrThrow(Queue_.Cluster),
+            Invoker_,
+            ProfilingCounters_,
+            Queue_.Path,
+            ExportConfig_,
+            DynamicConfig_,
+            Logger);
+    }
 
-    auto exportTask = New<TQueueExportTask>(
-        ClientDirectory_->GetClientOrThrow(Queue_.Cluster),
-        ProfilingCounters_,
-        Queue_.Path,
-        config,
-        Logger);
+    auto exportTaskError = WaitFor(exportTask->Run());
 
-    auto nextExportProgress = exportTask->Run();
+    auto newExportProgress = exportTask->GetExportProgress();
 
     auto guard = Guard(Lock_);
 
-    if (config.ExportDirectory == ExportConfig_.ExportDirectory) {
-        ExportProgress_ = std::move(nextExportProgress);
+    if (exportConfig->ExportDirectory == ExportConfig_->ExportDirectory) {
+        if (newExportProgress) {
+            ExportProgress_ = std::move(newExportProgress);
+        } else {
+            // NB(apachee): It is possible that task failed before creating updated export progress.
+            // XXX(apachee): Should we assert that exportTaskError is not OK in this case?
+            THROW_ERROR_EXCEPTION("Export task failed to create updated export progress, internal progress stayed the same")
+                << TErrorAttribute("TaskError", exportTaskError)
+                << TErrorAttribute("LastSubtaskError", exportTask->GetLastSubtaskError());
+        }
     }
+
+    guard.Release();
```

Comment:
My personal preference with guards is scoping. If you are not doing that, you might as well use plain `Lock_.Acquire()` and `Lock_.Release()`.
But I think we tend to prefer this:
```
{
    auto guard = Guard(Lock_);
    // Do something.
}
```